User/UserRepositoryCollection.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\User;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use SumoCoders\FrameworkMultiUserBundle\Exception\RepositoryNotRegisteredException;
+
+class UserRepositoryCollection
+{
+    /**
+     * @var ArrayCollection
+     */
+    private $userRepositories;
+
+    /**
+     * UserRepositoryCollection constructor.
+     *
+     * @param array $userRepositories
+     */
+    public function __construct(array $userRepositories)
+    {
+        $this->userRepositories = new ArrayCollection();
+        foreach ($userRepositories as $repository) {
+            $this->registerUserRepository($repository);
+        }
+    }
+
+    /**
+     * Registers the UserRepository to the UserRepositoryCollection.
+     *
+     * @param UserRepository $userRepository
+     */
+    public function registerUserRepository(UserRepository $userRepository)
+    {
+        $this->userRepositories->add($userRepository);
+    }
+
+    /**
+     * Get the userRepositories.
+     *
+     * @return ArrayCollection
+     */
+    public function all()
```

Comment:
in andere collections gaat `all` een array returnen. Ik zou dat hier dan ook doen.